### PR TITLE
Default to drop-and-create for DevServices

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This allows for hibernate to be used without configuration.

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
@Sanne in this case we know that we just started the container, so we know there is no schema there (unless they have used a custom image with the schema in it). I think this makes sense in terms of making this 'just work' out of the box.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java [hunk 1]
```diff
@@ -177,6 +180,55 @@ void includeArchivesHostingEntityPackagesInIndex(HibernateOrmConfig hibernateOrm
...
+        } else {
+            dsName = "quarkus.datasource." + config.defaultPersistenceUnit.datasource.get() + ".username";
+        }
+
+        if (ConfigProvider.getConfig().getOptionalValue(dsName, String.class).isEmpty()) {
+            if (devServicesResult.getConfig().containsKey(dsName)) {
+                if (ConfigProvider.getConfig().getOptionalValue("quarkus.hibernate-orm.database.generation", String.class)
+                        .isEmpty()) {
+                    LOG.info(
+                            "Setting quarkus.hibernate-orm.database.generation=drop-and-create to initialize DevServices managed database");
```
Comment (gsmet) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:204]:
```suggestion
                            "Setting quarkus.hibernate-orm.database.generation=drop-and-create to initialize Dev Services managed database");
```

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java [hunk 2]
```diff
@@ -177,6 +180,55 @@ void includeArchivesHostingEntityPackagesInIndex(HibernateOrmConfig hibernateOrm
...
+            if (entry.getValue().datasource.isEmpty()) {
+                dsName = "quarkus.datasource.jdbc.url";
+            } else {
+                dsName = "quarkus.datasource." + entry.getValue().datasource.get() + ".username";
+            }
+            if (ConfigProvider.getConfig().getOptionalValue(dsName, String.class).isEmpty()) {
+                if (devServicesResult.getConfig().containsKey(dsName)) {
+                    String propertyName = "quarkus.hibernate-orm." + entry.getKey() + ".database.generation";
+                    if (ConfigProvider.getConfig().getOptionalValue(propertyName, String.class).isEmpty()) {
+                        LOG.info("Setting " + propertyName + "=drop-and-create to initialize DevServices managed database");
```
Comment (gsmet) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:222]:
```suggestion
                        LOG.info("Setting " + propertyName + "=drop-and-create to initialize Dev Services managed database");
```

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRuntimeConfigPersistenceUnit.java [hunk 3]
```diff
@@ -72,6 +72,9 @@ public boolean isAnyPropertySet() {
          *
          * `drop-and-create` is awesome in development mode.
          *
+         * This defaults to 'none', however if DevServices is in use and no other extensions that manage the schema are present
```
Comment (gsmet) [inline extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRuntimeConfigPersistenceUnit.java:75]:
```suggestion
         * This defaults to 'none', however if Dev Services is in use and no other extensions that manage the schema are present
```

**Review by gsmet — COMMENTED** [review]
Added a minor gripe: that's how it's documented everywhere now so let's be consistent.

**Comment by Sanne** [comment]
Nice idea, thanks!

I wonder if the check on `JdbcDataSourceSchemaReadyBuildItem` is enough for getting a consistent experience though, as it's merely controlling the phase in which schema management needs to happen (for ordering), but it doesn't really tell

1. if schemas are actually being generated
2. which ones, in case of multiple datasources

I'll play a bit more with it to see if I'm missing something.

**Comment by geoand** [comment]
I was actually thinking about this yesterday when creating a demo. Definitely +1 from me for the idea.

**Comment by Sanne** [comment]
Ok looks like this isn't too tricky when there are multiple PUs, works fine even on non-default PUs - although obviously one is no longer able to run "fully configuration-less"

**Review by Sanne — APPROVED** [review]

**Pull request merged**